### PR TITLE
Deprecate play.Logger

### PR DESCRIPTION
**Pull request by benmccann**
Addresses https://github.com/playframework/playframework/issues/1669 for Play 2.7

### CODE REVIEW

**Comment by richdougherty** [comment]
If we deprecate this I think we want to mention the suggested migration in the migration guide too. Is `Logger` mentioned in our documentation?

**Comment by benmccann** [comment]
I've updated the docs. There's no migration guide for 2.7.0 yet. Do you want me to start a new one?

**Comment by richdougherty** [comment]
That would be great if you could start the migration guide, thanks.

**Comment by mkurz** [comment]
@benmccann Something is wrong with the build as well.

**Comment by benmccann** [comment]
Hey guys, sorry for the delay on getting back to this one. I've added a start to the migration guide and fixed the failing build

**Comment by marcospereira** [comment]
Small side effect here: `play.Logger` configures the logger name as `application`:

https://github.com/playframework/playframework/blob/2.6.3/framework/src/play/src/main/java/play/Logger.java#L43

And then users have the following line in `conf/logback.xml` configuration:

```xml
<logger name="application" level="DEBUG" />
```
The same happens with our default logs (and examples):

1. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-logSql.xml#L31
2. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-dev.xml#L16
3. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-default.xml#L20

So we can both remove this line from our `logback.xml` files and it may be worth adding a note in the migration guide.

**Comment by benmccann** [comment]
Good call. I've updated the migration guide to mention this

**Comment by marcospereira** [comment]
Thanks, @benmccann.

Sorry for nitpicking here, but do you mind also remove these lines since they are now useless:

1. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-logSql.xml#L31
2. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-dev.xml#L16
3. https://github.com/playframework/playframework/blob/2.6.3/framework/src/play-logback/src/main/resources/logback-play-default.xml#L20

**Comment by benmccann** [comment]
Thanks @marcospereira. Good catch! I've updated it

**Comment by marcospereira** [comment]
Thanks, @benmccann.

File: documentation/manual/releases/release27/migration27/Migration27.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!--- Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com> -->
+# Play 2.7 Migration Guide
+
+This is a guide for migrating from Play 2.6 to Play 2.7. If you need to migrate from an earlier version of Play then you must first follow the [[Play 2.6 Migration Guide|Migration26]].
+
+### `play.Logger` deprecated
+
+`play.Logger` has been deprecated in favor of using SLF4J directly. You can create an SLF4J logger with `private static final Logger logger = LoggerFactory.getLogger(YourClass.class);`. If you'd like a more concise solution, you may also consider [Project Lombok's `@Slf4j` annotation](https://projectlombok.org/features/log).
+
+If you have a in your logback.xml referencing the `application` logger, you may remove it.
```
Comment (PromanSEW) [inline documentation/manual/releases/release27/migration27/Migration27.md]: `If you have a` ... what? `in your...`

**Comment by mkurz** [comment]
Why was `play.Logger` deprecated but `play.api.Logger` was not?

**Pull request merged**